### DISCUSSION TITLE
use existing redis via socket

### DISCUSSION THREAD

**Discussion by thrillfall** [bookwyrm-social/bookwyrm#2132] [Q&A] [answered]
I am using the docker-compose setup and have an existing redis (cache for mastodon, nextcloud). How can i connect to the existing redis via `unix:///var/run/redis/redis-server.sock`? I mapped the socket file as volume which works nicely for my mastodon (also docker-compose) setup.

when i set `REDIS_ACTIVITY_HOST=unix:///var/run/redis/redis-server.sock`
i get 
``` ValueError: Database is int between 0 and limit - 1, not var/run/redis/redis-server.sock:0/2
flower_1| [E ...] Failed to capture events: 'Database is int between 0 and limit - 1, not var/run/redis/redis-server.sock:0/2', trying again in 8 seconds.```

**Comment by thrillfall** [comment]
i found there is a notation for socket connection:
now i have 
`REDIS_ACTIVITY_HOST=redis+socket:///var/run/redis/redis-server.sock?virtual_host=1`
and i get

```File "/usr/local/lib/python3.9/site-packages/kombu/utils/url.py", line 51, in parse_url
celery_worker_1  |     return dict(transport=scheme, hostname=host,
celery_worker_1  | TypeError: dict() got multiple values for keyword argument 'virtual_host'
bookwyrm_celery_beat_1 exited with code 1```

**Comment by jaschaurbach** [comment, marked as answer]
This will be solved once #2604 is ready
